substrate/frame/identity/src/lib.rs

```diff
@@ -97,6 +100,9 @@ type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
 >>::NegativeImbalance;
 type AccountIdLookupOf<T> = <<T as frame_system::Config>::Lookup as StaticLookup>::Source;
 
+/// Maximum size of an identity we can store is 4 MiB.
+const IDENTITY_MAX_SIZE: u32 = 4 * 1024 * 1024;
```

Comment:
This seems very large, I imagine even 1 kB would be enough. Maybe 5 kB?

Reply:
I just took the constant from the preimage pallet. Here is a more informed view: a "full" `IdentityInfo` without any entries in `additional` has an encoded size of 253 bytes. Each entry in `additional` is then worth at most 66 bytes, depending on the values chosen for the key and value, since they're `Data`. With the current values for `T::MaxAdditionalFields` of 100, that brings us to at most `253 + 66 * 100` or ~6.7 KiB maximum encoded size for any value currently in storage. Considering this, I set it to 7 KiB now and we can lower it later if we wish to do so.

Reply:
Yeah, that sounds good. The preimage pallet is meant to store data such that you can remove it if the system hasn't requested it. Setting an identity clears any judgements, so we don't really expect people to remove their data once they have set it. Of course the byte fee accounts for it, but it seems more flexible to just have a field meant to take a hash that can be hosted elsewhere. Then, people could actually use the preimage pallet if it needed to be on-chain, or some off-chain hosting service if the reader of the Identity Info knew where to look for it.